File: 53. Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxA = nums[0];
    let sum = 0;
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > sum + nums[i]) {
            sum = nums[i];
        } else {
            sum += nums[i];
        }
        maxA = Math.max(maxA, sum);
    }
    
    return maxA;
};

File: 53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total = 0
        max_sub = nums[0]
        
        for i in range(len(nums)):
            if nums[i] > total + nums[i]:
                total = nums[i]
            else:
                total += nums[i]
            
            max_sub = max(max_sub, total)
        
        return max_sub
